src/components/Content/PDFButton.js

```diff
@@ -28,19 +28,25 @@ class PDFButton extends React.Component {
     parentPage.children.forEach((page) => this.buildPageIds(pageIds, page))
   }
 
-  fetchUrl () {
-    this.setState(Object.assign({}, this.state, {loading: true}))
+  componentWillUpdate (nextProps, nextState) {
```

Comment:
Is this needed? Is this an optimization?

Reply:
Should the generation stop if you change location? Do we need to care about this?

Reply:
Yup, we definitely should, because then the pdf button should obviously load the pdf for a different location.
Since I've put the PDF-Button somewhere else, it is not detached and attached every time, but reused when changing location.

Reply:
Maybe we should move it then back to the component where is belongs. So react takes care of resetting state etc.

Reply:
Nope, doesn't belong there. Also a component should react to the properties it gets passed through, the fact that it worked before was just luck